### ISSUE TITLE
Request: Lock Size/Position of popped-out tabs + Coded Positioning/Size

### ISSUE THREAD

**Issue by BlackWyvern** [Infiziert90/ChatTwo#88]
Using the "Allow moving chat" and "Allow resizing chat" is disregarded for popped out tabs.

It would be nice if an option existed to lock said popouts and to pixel position them similarly to the master chat tab.

The ability to pixel resize all tabs would be nice, but might be a bit too extra.

**Comment by codeinclined** [comment]
I am considering contributing some improvements to the popout tabs and can have this be one of things I look into doing as it's something I've wanted as well. Not making any promises as I'm just now starting to look at the codebase! 😄

**Comment by codeinclined** [comment]
Actually I just had an epiphany (maybe being overly dramatic? lol). Chat 2 works REALLY well with window docking! Try going into `/xlsettings` and enabling docking then drag the popouts over the main window or other popouts to get a docking interface. You can have them stack that way as a group one on top of another with the same width or as tabs which is _probably_ what you're looking for.

I was spending quite a bit of time rewriting the popout code to have popouts have their own tabs, but didn't know Dalamud / ImGUI natively support basically the same thing. Maybe give that a try? @Infiziert90 might be good to draw attention to the docking feature in the documentation or something as it's really useful for this plugin?

**Comment by Infiziert90** [comment]
> Actually I just had an epiphany (maybe being overly dramatic? lol). Chat 2 works REALLY well with window docking! Try going into `/xlsettings` and enabling docking then drag the popouts over the main window or other popouts to get a docking interface. You can have them stack that way as tabs or as a group one on top of another with the same width which is _probably_ what you're looking for.
> 
> I was spending quite a bit of time rewriting the popout code to have popouts have their own tabs, but didn't know Dalamud / ImGUI natively support basically the same thing. Maybe give that a try? @Infiziert90 might be good to draw attention to the docking feature in the documentation or something as it's really useful for this plugin?
> 
> TL;DR disregard my comment about me contributing changes and try out the docking feature. After finding that out my changes are no longer necessary I've decided not to continue with those

The docking feature overall is a big mess, and has lead to multiple issues in the past ... so yeah, not the most fond of it

**Comment by abbeynels** [comment]
I'd also appreciate if we could lock popouts as well. When I right click over an item in chat, the popup often is cut off by the bottom of my screen. Here's a screenshot: 
![chat2](https://github.com/user-attachments/assets/0a2f6f8c-256b-4b81-b15b-b77862fd1ebc)

**Comment by Infiziert90** [comment]
Implement in 97a8b96326020459f5ea7ab306cc645a67626b75
Position+Size as sliders isn't on the list of planned things, just too complicated

And @abbeynels this is a different issue, popouts are your actual chat tabs in a separated window
i will still look into why this keeps happening
